src/main/java/mekanism/client/render/item/ChemicalFluidBarDecorator.java

```diff
@@ -0,0 +1,116 @@
+        for (Capability<? extends IChemicalHandler<?,?>> chemicalCap : chemicalCaps) {
+            Optional<? extends IChemicalHandler<?, ?>> capabilityInstance = stack.getCapability(chemicalCap).resolve();
+            if (capabilityInstance.isPresent()) {
+                IChemicalHandler<?, ?> chemicalHandler = capabilityInstance.get();
+                if (chemicalHandler.getTanks() == 0) {
+                    continue;
+                }
+                int tank = 0;
+                if (chemicalHandler.getTanks() > 1 && Minecraft.getInstance().level != null) {
+                    //Cycle through multiple tanks every second, to save some space if multiple tanks are present
+                    tank = (int) (Minecraft.getInstance().level.getGameTime() / 20) % chemicalHandler.getTanks();
+                }
+                renderBar(bufferBuilder, xOffset, yOffset, chemicalHandler.getChemicalInTank(tank), chemicalHandler.getTankCapacity(tank));
+                yOffset--;
+            }
+        }
+
+        if (showFluid) {
+            Optional<IFluidHandlerItem> capabilityInstance = FluidUtil.getFluidHandler(stack).resolve();
+            if (capabilityInstance.isPresent()) {
+                IFluidHandlerItem fluidHandler = capabilityInstance.get();
+                if (fluidHandler.getTanks() != 0) {
+                    int tank = 0;
+                    if (fluidHandler.getTanks() > 1 && Minecraft.getInstance().level != null) {
+                        //Cycle through multiple tanks every second, to save some space if multiple tanks are present
+                        tank = (int) (Minecraft.getInstance().level.getGameTime() / 20) % fluidHandler.getTanks();
+                    }
+                    renderBar(bufferBuilder, xOffset, yOffset, fluidHandler.getFluidInTank(tank), fluidHandler.getTankCapacity(tank));
+                }
+            }
+        }
+        return true;
+    }
+
+    private void renderBar(BufferBuilder bufferBuilder, int stackXPos, int yPos, ChemicalStack<?> stack, long capacity) {
+        int color = stack.getChemicalColorRepresentation();
+        renderBar(bufferBuilder, stackXPos, yPos, (float)((double) stack.getAmount() / capacity), color);
+    }
+
+    private void renderBar(BufferBuilder bufferBuilder, int stackXPos, int yPos, FluidStack stack, long capacity) {
+        int color = IClientFluidTypeExtensions.of(stack.getFluid()).getTintColor(stack);
+        renderBar(bufferBuilder, stackXPos, yPos, (float)((double) stack.getAmount() / capacity), color);
+    }
+
+    private void renderBar(BufferBuilder bufferBuilder, int stackXPos, int yPos, float width, int color) {
+        ItemRenderer itemRenderer = Minecraft.getInstance().getItemRenderer();
+        itemRenderer.fillRect(bufferBuilder, stackXPos + 2, yPos, 13, 1, 0, 0, 0, 255);
+        itemRenderer.fillRect(bufferBuilder, stackXPos + 2, yPos, convertWidth(width), 1, getRed(color), getGreen(color), getBlue(color), 255);
```

Comment:
I am guessing this is done like vanilla did it but is there a reason to not have the two draws be:
```java
int barWidth = convertWidth(width);
fill(..., stackXPos + 2, yPos, barWidth, 1, color);
int emptyWidth = 13 - barWidth;
if (emptyWidth > 0) {
    fill(..., stackXPos + 2 + barWidth, yPos, 13 - barWidth, 1, black);
}
```

And if we do end up using `GuiUtils#fill`, could also just add in validation to that which makes it NO-OP if width or height is zero so that some of the above can just be inlined for drawing the black part